db/views/reporting_facility_states_v01.sql

```diff
@@ -0,0 +1,77 @@
+WITH
+    registered_patients AS (
+        SELECT registration_facility_region_id as region_id, month_date,
+               COUNT(*) as cumulative_registrations,
+               COUNT(*) FILTER (WHERE months_since_registration = 0) as monthly_registrations
+        FROM reporting_patient_states
+        WHERE hypertension = 'yes'
+        GROUP BY 1, 2
+    ),
+
+    assigned_patients AS (
+        SELECT assigned_facility_region_id as region_id, month_date,
+               COUNT(distinct(patient_id)) FILTER (WHERE htn_care_state = 'under_care') as under_care,
+               count(distinct(patient_id)) FILTER (WHERE htn_care_state = 'lost_to_follow_up') as lost_to_follow_up,
+               count(distinct(patient_id)) FILTER (WHERE htn_care_state = 'dead') as dead,
+               COUNT(*) as assigned_patients
+        FROM reporting_patient_states
+        WHERE hypertension = 'yes'
+        GROUP BY 1, 2
+    ),
+
+    treatment_outcomes_in_last_3_months AS (
+        SELECT assigned_facility_region_id as region_id, month_date,
+
+               count(distinct(patient_id)) FILTER (WHERE htn_treatment_outcome_in_last_3_months = 'controlled' AND htn_care_state = 'under_care') as controlled_under_care,
+               count(distinct(patient_id)) FILTER (WHERE htn_treatment_outcome_in_last_3_months = 'controlled' AND htn_care_state = 'lost_to_follow_up') as controlled_lost_to_follow_up,
+
+               count(distinct(patient_id)) FILTER (WHERE htn_treatment_outcome_in_last_3_months = 'uncontrolled' AND htn_care_state = 'under_care') as uncontrolled_under_care,
+               count(distinct(patient_id)) FILTER (WHERE htn_treatment_outcome_in_last_3_months = 'uncontrolled' AND htn_care_state = 'lost_to_follow_up') as uncontrolled_lost_to_follow_up,
+
+               count(distinct(patient_id)) FILTER (WHERE htn_treatment_outcome_in_last_3_months = 'missed_visit' AND htn_care_state = 'under_care') as missed_visit_under_care,
+               count(distinct(patient_id)) FILTER (WHERE htn_treatment_outcome_in_last_3_months = 'missed_visit' AND htn_care_state = 'lost_to_follow_up') as missed_visit_lost_to_follow_up,
+
+               count(distinct(patient_id)) FILTER (WHERE htn_treatment_outcome_in_last_3_months = 'visited_no_bp' AND htn_care_state = 'under_care') as visited_no_bp_under_care,
+               count(distinct(patient_id)) FILTER (WHERE htn_treatment_outcome_in_last_3_months = 'visited_no_bp' AND htn_care_state = 'lost_to_follow_up') as visited_no_bp_lost_to_follow_up,
+
+               count(distinct(patient_id)) FILTER (WHERE htn_care_state = 'under_care') as patients_under_care,
+               count(distinct(patient_id)) FILTER (WHERE htn_care_state = 'lost_to_follow_up') as patients_lost_to_follow_up
+         FROM reporting_patient_states
+        WHERE hypertension = 'yes'
+          AND months_since_registration >= 3
+        GROUP BY 1, 2
+    )
+
+SELECT
+cal.*,
+rf.*,
```

Comment:
Yeah, perhaps. Like you're saying, it depends on how well the facility states view works, and if it will suffice for all our dashboard needs. I'd like to also consider usability of the patient states view via metabase, and how comfortable people would be with joining tables to get information like facility-name, quarter-string, etc.

Joining dimension tables (as opposed to fact tables) is cheap enough to do as per need.

Reply:
I'd be comfortable enough needing to join dimension tables to enrich results. That would be expensive if we want to join patient states with facilities, right? It would be cheap if we were joining facility states with facilities, or if we made sure to aggregate our one-off reports to per-facility before joining?